user_test.go

```diff
@@ -103,6 +103,34 @@ func TestListUser(t *testing.T) {
 	autopilot.Equals(t, ol.UserRoleAdmin, result[2].Role)
 }
 
+func TestListUserOmitDeactivated(t *testing.T) {
+	// Arrange
+	testRequestOne := autopilot.NewTestRequest(
+		`query UserList($after:String!$filter:[UsersFilterInput!]$first:Int!){account{users(after: $after, first: $first, filter: $filter){nodes{id,email,htmlUrl,name,role},{{ template "pagination_request" }},totalCount}}}`,
+		`{ "after": "", "filter": [ {"key": "deactivated_at", "type": "equals"} ], "first": 100 }`,
+		`{ "data": { "account": { "users": { "nodes": [ {{ template "user_1" }}, {{ template "user_2" }} ], {{ template "pagination_initial_pageInfo_response" }}, "totalCount": 2 }}}}`,
+	)
+	testRequestTwo := autopilot.NewTestRequest(
+		`query UserList($after:String!$filter:[UsersFilterInput!]$first:Int!){account{users(after: $after, first: $first, filter: $filter){nodes{id,email,htmlUrl,name,role},{{ template "pagination_request" }},totalCount}}}`,
+		`{ "after": "OA", "filter": [ {"key": "deactivated_at", "type": "equals"} ], "first": 100 }`,
+		`{ "data": { "account": { "users": { "nodes": [], {{ template "pagination_second_pageInfo_response" }}, "totalCount": 0 }}}}`,
+	)
+	requests := []autopilot.TestRequest{testRequestOne, testRequestTwo}
+
+	client := BestTestClient(t, "user/list_omit_deactivated", requests...)
+	// Act
+	payloadVars := client.InitialPageVariablesPointer().WithoutDeactivedUsers()
```

Comment:
While this is not [functional options](https://golang.cafe/blog/golang-functional-options-pattern.html) - i don't hate it and it is better then before.

Reply:
I started testing out the functional options approach but think the builder pattern is more appropriate in this case.
We already have `client.InitialPageVariables()` and `client.InitialPageVariablesPointer()`. Adding some form of `New()` without a client, or `New(client, ....)` seems like bloat

Reply:
👍